### DISCUSSION TITLE
problem with the example of the quantum espresso calculator for Gold

### DISCUSSION THREAD

**Discussion by Tongca** [SSCHAcode/python-sscha#143] [Q&A] [answered]
Dear Developers of SSCHA,

When I do a simulation in the NVT ensemble using the script from sscha_and_dft example, I am facing the below error message:
[Au_output.txt](https://github.com/SSCHAcode/python-sscha/files/12144010/Au_output.txt)
Please help me to understand what's going wrong.

The input is as follows：

import sscha, sscha.Ensemble, sscha.SchaMinimizer, sscha.Relax, sscha.Utilities
import cellconstructor as CC, cellconstructor.Phonons
import cellconstructor.Structure, cellconstructor.calculators
import cellconstructor.calculators
from ase.calculators.espresso import Espresso
import numpy as np, matplotlib.pyplot as plt
import sys, os

input_data = {
    'control' : {
        'disk_io' : 'None',
        'pseudo_dir' : '.'
    },
    'system' : {
        'ecutwfc' : 45,   # Cutoff for wavefunction
        'ecutrho' : 180, # Cutoff for the density
        'occupations' : 'smearing',
        'smearing' : 'mv',
        'degauss' : 0.03
    },
    'electrons' : {
        'conv_thr' : 1e-4
    }
}
pseudopotentials = {'Au' : 'Au_ONCV_PBE-1.0.oncvpsp.upf'}

kpts = (1,1,1)
koffset = (1,1,1)

command = 'mpirun -np 24 pw.x -npool 1 -i PREFIX.pwi > PREFIX.pwo'

calculator = Espresso(input_data = input_data,
                                   pseudopotentials = pseudopotentials,
                                   command = command,
                                   kpts = kpts,
                                   koffset = koffset)

TEMPERATURE = 300
N_CONFIGS = 50
MAX_ITERATIONS = 20
START_DYN = 'start_dyn'
NQIRR = 13

gold_dyn = CC.Phonons.Phonons(START_DYN, NQIRR)

ensemble = sscha.Ensemble.Ensemble(gold_dyn, TEMPERATURE)

minim = sscha.SchaMinimizer.SSCHA_Minimizer(ensemble)
minim.set_minimization_step(0.01) 

relax = sscha.Relax.SSCHA(minim, calculator, N_configs = N_CONFIGS,
                          max_pop = MAX_ITERATIONS)

ioinfo = sscha.Utilities.IOInfo()
ioinfo.SetupSaving("minim_info")
relax.setup_custom_functions(custom_function_post = ioinfo.CFP_SaveAll)

relax.relax(get_stress = True)

relax.minim.finalize()
relax.minim.dyn.save_qe("sscha_T{}_dyn".format(TEMPERATURE))

Best wishes.

**Comment by mesonepigreco** [comment, marked as answer]
Hi, your espresso calculator is not printing the forces in output, which are required for the SSCHA minimization.
Try to add the forces (and maybe also the stresses if you later want to do a variable cell relaxation) by adding them to the control key of the espresso input data:

```
'control' : {
'disk_io' : 'None',
'pseudo_dir' : '.',
'tprnfor' : True,   # Print forces on output
'tstress': True     # Print the stress on output
}, 
```

I hope this solves the problem,
Bests,
Lorenzo

P.S.
Thanks for spotting the error in the example. Probably it was working with an older version of ASE. I have updated and in will be merged soon
